### ISSUE TITLE
etl travis deploy is broken

### ISSUE THREAD

**Issue by stephen-soltesz** [m-lab/etl#532]
The last three commits to master are failing with messages like:

```
DEBUG: Running command: [/home/travis/google-cloud-sdk/bin/gsutil cp /tmp/tmpua8Upc/fun.zip gs://functions-mlab-sandbox/us-central1-projects/mlab-sandbox/locations/us-central1/functions/embargoOnFileNotificationSandbox-tnueomgruijb.zip]]
DEBUG: Executing command: [u'/home/travis/google-cloud-sdk/bin/gsutil', u'cp', u'/tmp/tmpua8Upc/fun.zip', u'gs://functions-mlab-sandbox/us-central1-projects/mlab-sandbox/locations/us-central1/functions/embargoOnFileNotificationSandbox-tnueomgruijb.zip']
DEBUG: (gcloud.beta.functions.deploy) ResponseError: status=[403], code=[Forbidden], message=[The caller does not have permission]
Traceback (most recent call last):
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/cli.py", line 844, in Execute
    resources = calliope_command.Run(cli=self, args=args)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/backend.py", line 756, in Run
    resources = command_instance.Run(args)
  File "/home/travis/google-cloud-sdk/lib/surface/functions/deploy.py", line 153, in Run
    return _Run(args)
  File "/home/travis/google-cloud-sdk/lib/surface/functions/deploy.py", line 116, in _Run
    return api_util.PatchFunction(function, updated_fields)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/api_lib/functions/util.py", line 286, in CatchHTTPErrorRaiseHTTPExceptionFn
    base_exceptions.HttpException(GetHttpErrorMessage(error)))
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/exceptions.py", line 137, in reraise
    six.reraise(type(exc_value), exc_value, tb)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/api_lib/functions/util.py", line 283, in CatchHTTPErrorRaiseHTTPExceptionFn
    return func(*args, **kwargs)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/api_lib/functions/util.py", line 335, in PatchFunction
    updateMask=fields_to_patch_str,
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/third_party/apis/cloudfunctions/v1/cloudfunctions_v1_client.py", line 352, in Patch
    config, request, global_params=global_params)
  File "/home/travis/google-cloud-sdk/lib/third_party/apitools/base/py/base_api.py", line 730, in _RunMethod
    return self.ProcessHttpResponse(method_config, http_response, request)
  File "/home/travis/google-cloud-sdk/lib/third_party/apitools/base/py/base_api.py", line 736, in ProcessHttpResponse
    self.__ProcessHttpResponse(method_config, http_response, request))
  File "/home/travis/google-cloud-sdk/lib/third_party/apitools/base/py/base_api.py", line 603, in __ProcessHttpResponse
    http_response, method_config=method_config, request=request)
HttpException: ResponseError: status=[403], code=[Forbidden], message=[The caller does not have permission]
ERROR: (gcloud.beta.functions.deploy) ResponseError: status=[403], code=[Forbidden], message=[The caller does not have permission]
```

**Comment by yachang** [comment]
Any travis build link for this issue?

**Comment by stephen-soltesz** [comment]
@yachang https://travis-ci.org/m-lab/etl/branches

**Comment by stephen-soltesz** [comment]
At one time, the cloud function operations didn't support IAM well -- https://github.com/m-lab/etl/blob/integration/.travis.yml#L13 -- have you successfully gotten a service account to perform a cloud function deployment? That may still not be supported without overly-generous permissions to the travis service account.

**Comment by yachang** [comment]
It used to run successflly on mlab-sandbox:

https://travis-ci.org/m-lab/etl/builds/394208889  ss-sandbox deployment
https://travis-ci.org/m-lab/etl/builds/393806537 sandbox deployment

But since I merge the PR into integration branch, mlab-staging never deployed with success  even though I set the same role permission for mlab-staging and mlab-sandbox 

The worse thing now is that I create a new branch ss-sandbox-cloud2 (without any code change) from integration, and it cannot deploy on mlab-sandbox now :(

**Comment by yachang** [comment]
Thanks to Peter who read EVERY email, we found that gcloud team changed policy on service account running cloud function.

 GCF will require the IAM Service Account User role (roles/iam.serviceaccountuser) in addition to the existing Cloud Functions Developer role (roles/cloudfunctions.developer) in order to manage functions. If a user does not have both of these roles, they will not be able to create, update, or delete functions. Running functions will not be affected.

Some updates: serviceaccountuser role is under service account category, not iam category.

Once new role was added, the mlab-sandbox deployment working and mlab-staging is expected to work as well.
